### ISSUE TITLE
When last window does not take focus, global shortcuts don't work any more

### ISSUE THREAD

**Issue by sthibaul** [mate-desktop/marco#703]
#### Steps to reproduce the behaviour

* Run the MATE desktop with the Marco window manager
* Do not open any application except the following
* Open a mate-terminal, open two windows
* Run this python script in the first window
[test.py.txt](https://github.com/mate-desktop/marco/files/7523644/test.py.txt)
* Reduce this first window, focus switches to the second mate-terminal window
* exit the second mate-terminal window
* Press alt-F1

#### Expected behaviour

The application menu should open

#### Actual behaviour

The application menu does not open

#### Details

Normally (without running the attached script), when the last window is closed, Marco sets the focus to the Desktop window.

When running the attached script, this however fails. The attached script creates a window that has the gtk `accept-focus` property to `False`, i.e. with `XWMHints.input` set to `False`. When exiting the second mate-terminal window, marco tries to switch back to the last window, i.e. the window of the script, but since it has the input hint set to false, marco fails to do so, and apparently doesn't set the focus to the Desktop window, and instead it's the root window that gets the focus. And then the global shortcuts such as alt-F1 do not work any more.

The actual user case at stake is people using the Orca screen reader with the braille monitor enabled (`orca -e braille-monitor`), which disables input the same way as the attached script. Whenever they close the last application and thus the braille monitor is the last window, the focus is stuck on the root window. And they cannot use a global shortcut to get back focus to anything useful, i.e. they end up being completely stuck and have to reboot the system...

#### MATE general version

1.24.1

#### Package version

1.24.1

#### Linux Distribution

Debian

**Comment by mmhere** [comment]
https://github.com/mate-desktop/marco/issues/647

related?

Lots of focus problems in releases later than 1.20.1.

**Comment by ethindp** [comment]
I've faced this issue too, and this is the (major) reason that many (and I mean many) blind users explicitly avoid Mate and tell others to not use it: this bug is definitely not minor (or at least I don't consider it one), and it has caused me much frustration and irritation, especially since this bug has been open for almost a year and I'd expect something to come of it. I've disabled braille in Orca, but I don't know if that actually disables the braille monitor. This appears to most often happen with Electron applications, though other applications trigger it as well. I'm using Arch Linux, latest Mate release.

**Comment by vojtapolasek** [comment]
I agree. I am also a blind user of Mate desktop. I really like the desktop because it is fast and lightweight, but this bug is really annoying and hindering the usability. The question from new linux users: "But this just does not happen in Windows, why here?" is really not easy to answer... Please investigate this. Thank you.
